packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -166,6 +166,30 @@ Future<int> runCommandAndStreamOutput(List<String> cmd, {
   return await process.exitCode;
 }
 
+/// Runs the [command] interactively, connecting the stdin/stdout/stderr
+/// streams of this process to those of the child process. Completes with
+/// the exit code of the child process.
+Future<int> runInteractively(List<String> command, {
+  String workingDirectory,
+  bool allowReentrantFlutter: false,
+  Map<String, String> environment
+}) async {
+  final Process process = await runCommand(
+    command,
+    workingDirectory: workingDirectory,
+    allowReentrantFlutter: allowReentrantFlutter,
+    environment: environment,
+  );
+  process.stdin.addStream(stdin);
+  final Future<dynamic> stdOutDone = stdout.addStream(process.stdout);
+  final Future<dynamic> stdErrDone = stderr.addStream(process.stderr);
+  // Wait for stdout and stderr to be fully processed, because process.exitCode
+  // may complete first.
+  await stdOutDone;
+  await stdErrDone;
```

Comment:
Probably best to wait for both in a Future.wait in case they're interlocked somehow (not sure how buffering works here).